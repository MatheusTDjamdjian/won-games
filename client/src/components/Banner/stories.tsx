import React from 'react'
import { Meta, StoryFn} from '@storybook/nextjs'

import Banner from '.'

import { BannerProps } from './types'

export default {
  title: 'Banner',
  component: Banner,
  argTypes: {
    ribbon: {
      type: 'string'
    }
  },
  args: {
    img: 'https://cdn.pixabay.com/photo/2023/09/27/12/15/river-8279466_1280.jpg',
    title: 'Defy death',
    subtitle: '<p>Play the new <strong>CrashLands</strong> season',
    buttonLabel: 'Buy now',
    buttonLink: '/games/defy-death'
  },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: StoryFn<BannerProps> = (args: BannerProps) => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Banner {...args} />
  </div>
)

export const WithRibbon: StoryFn<BannerProps> = (args: BannerProps) => (
  <div style={{ maxWidth: '104rem', margin: '0 auto' }}>
    <Banner {...args} />
  </div>
)

WithRibbon.args = {
  ribbon: '20% OFF',
  ribbonSize: 'normal',
  ribbonColor: 'primary'
}