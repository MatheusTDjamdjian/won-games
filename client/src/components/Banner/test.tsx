import React from 'react'
import { screen } from '@testing-library/react'

import Banner from '.'
import { renderWithTheme } from '@/utils/tests/helpers'

const props = {
  img: 'https://cdn.pixabay.com/photo/2023/09/27/12/15/river-8279466_1280.jpg',
  title: 'Defy death',
  subtitle: '<p>Play the new <strong>CrashLands</strong> season',
  buttonLabel: 'Buy now',
  buttonLink: '/games/defy-death'
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Banner {...props} />)

    expect(
      screen.getByRole('heading', { name: /Defy death/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Play the new CrashLands season/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /Defy death/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a Ribbon', () => {
    renderWithTheme(<Banner {...props}  ribbon="My Ribbon" ribbonSize="small" ribbonColor="secondary"/>)

    const ribbon = screen.getByText(/My Ribbon/i)

    expect(ribbon).toBeInTheDocument()
    expect(ribbon).toHaveStyle({ backgroundColor: '#3CD3C1'})
    expect(ribbon).toHaveStyle({
      height: '2.6rem',
      fontSize: '1.2rem',
    })
  })
})