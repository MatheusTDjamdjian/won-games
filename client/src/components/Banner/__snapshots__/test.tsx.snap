// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
<main
  class="sc-fYsHOw jlUmQM"
>
  <div
    aria-label="Defy death"
    class="sc-dKREkF cGsEZZ"
    role="img"
    src="https://cdn.pixabay.com/photo/2023/09/27/12/15/river-8279466_1280.jpg"
  />
  <div
    class="sc-fWnslK kOgSrX"
  >
    <h2
      class="sc-dIMoHT dRAqgO"
    >
      Defy death
    </h2>
    <h3
      class="sc-iQQCXo jRwqiS"
    >
      <p>
        Play the new 
        <strong>
          CrashLands
        </strong>
         season
      </p>
    </h3>
    <a
      class="sc-gDpztx fQSLNn"
      href="/games/defy-death"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</main>
`;
